### PR TITLE
Seed the tiny model generation scripts

### PR DESCRIPTION
**Pull request by albertvillanova**
Nothing under `scripts/generate_tiny_models/` seeds the RNG, so every run of a generation script produces different weights. Re-pushing a tiny model whose recipe has not changed therefore always writes a new `model.safetensors`.

That defeats a property `huggingface_hub` already provides: `create_commit` drops files that are unchanged and short-circuits an all-unchanged commit with "No files have been modified since last commit", opening no PR at all. The tokenizer, config and model card already benefit from this on a re-push. The weights never do, purely because the initialization is unseeded, so a config-only change produces a Hub diff that also churns the weights, and a no-op regeneration still opens a PR.

It also blocks verification. Writing a generation script for a tiny model that already exists cannot be checked by running it, because the run produces a different artefact rather than reproducing the published one.

### Solution

A shared `SEED` and a `set_seed()` helper in `_common.py`, called by every script directly after `check_transformers_version()`.

The call has to come before the model is constructed, not before `init_weights_tiny_model`. Only 39 of the scripts call that helper at all, and in `for_conditional_generation` it is 2 of 20, so for most of them every draw has already happened inside the model's own initialization by the time it would run.

### Scope

- `_common.py`: `SEED` and `set_seed()`
- 61 generation scripts: one `set_seed()` call each
- `README.md`: a `Seeding` section, plus the step in the numbered list and in "Adding a new tiny model"

No tiny model is regenerated here, so nothing on the Hub changes. The effect starts at the next regeneration, and the first one after this still rewrites the weights once, since the published artefacts came from unseeded runs.

### A limit worth recording

The seed fixes the RNG stream, not the library producing it. `check_transformers_version` pins transformers exactly, but `nn.init` belongs to torch, with no upper bound, so a torch upgrade can still change the weights.

---

> [!NOTE]
> **Low Risk**
> Tooling-only change to test fixture generation scripts; no runtime TRL library behavior changes, though the next regeneration will one-time update Hub weight blobs.
> 
> **Overview**
> Adds **deterministic RNG seeding** to the tiny-model Hub generation pipeline so re-running a script without recipe changes no longer churns `model.safetensors` and spurious Hub PRs.
> 
> `_common.py` introduces shared `SEED` (42) and `set_seed()`, which wraps `transformers.set_seed`. **Every generation script** now calls `set_seed()` immediately after `check_transformers_version()` and **before** model construction (documented in README with a new **Seeding** section and updated workflow steps). `peft_qwen3_for_causal_lm_2.py` uses `set_seed(SEED + 1)` so its LoRA weights stay distinct from the sibling PEFT script.
> 
> **No Hub artifacts are regenerated in this PR**; the first post-merge regeneration will still rewrite weights once because existing checkpoints were built unseeded. Docs note that torch is not pinned, so a torch upgrade can still change weights despite a fixed seed.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bdd6df7cb0100f8b5867ee373b6e72b1433e6b2a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by AmineDiro — APPROVED** [review]
Nice 👍🏼 

question: does `transformers._set_seed()` handle all init functions for weights ?

File: scripts/generate_tiny_models/for_causal_lm/peft_qwen3_for_causal_lm_2.py [hunk 1]
```diff
@@ -17,11 +17,13 @@
 from peft import LoraConfig, get_peft_model
 from transformers import Qwen3ForCausalLM
 
-from .._common import check_transformers_version, push_to_hub, smoke_test
+from .._common import check_transformers_version, push_to_hub, set_seed, smoke_test
 
 
 check_transformers_version()
 
+set_seed()
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply scripts/generate_tiny_models/for_causal_lm/peft_qwen3_for_causal_lm_2.py]: Good catch, fixed in bdd6df7.

`peft_qwen3_for_causal_lm_2.py` says in its own first line that it exists to publish "different (random) LoRA weights" from `peft_qwen3_for_causal_lm.py`, and the only thing producing that difference was the unseeded RNG. A shared seed would have made the next regeneration of the two adapters byte-identical, so `tiny-PeftModel-2` would stop being a second adapter at all.

`set_seed` now takes an override and the second script passes `SEED + 1`. I checked the rest of the directory for the same pattern: no other script depends on differing randomness, and every other repeated suffix (`0.1`, `2.5`, `3.1`, ...) is a distinct reference model version rather than a re-roll of the same one.

One correction to the stated impact, since it matters for anyone reading this later. `test_with_multiple_adapters` cannot distinguish the two adapters today either, for an unrelated reason: a default `LoraConfig()` initializes `lora_B` to zeros, so a freshly created, never-trained adapter contributes nothing to the output. I pulled both published `adapter_model.safetensors` and compared them: `lora_A` differs between the two repos, `lora_B` is all zeros in both. So `expected_1` and `expected_2` in that test are already equal, and the assertions pass whether or not `use_adapter` actually switches anything. That is worth addressing, but separately, and it is not something this PR changes in either direction.

**Comment by albertvillanova** [comment]
Thanks. Short answer: yes for weight initialization, and the helper is `transformers.set_seed`, aliased privately here only to avoid clashing with the wrapper's name.

`transformers.set_seed` seeds `random`, `numpy.random`, `torch.manual_seed` and `torch.cuda.manual_seed_all`, plus the manual_seed_all of the mlu/musa/npu/hpu/xpu backends. Weight init in transformers goes through `torch.nn.init` and `torch.Tensor.normal_`, which draw from the global torch RNG that `torch.manual_seed` sets, so every init function is covered by construction. It is seeded per process before the model is built, so it also covers the models whose `_init_weights` runs during `from_pretrained` rather than through our `init_weights_tiny_model`.

Two limits worth stating. It does not pass `deterministic=True`, so `torch.use_deterministic_algorithms` stays off, but that affects kernel selection in ops like scatter and cuDNN convolutions, not the RNG draws that produce the weights. And it fixes the RNG stream, not the library producing it: `check_transformers_version` demands an exact transformers version, while torch only has a floor, so a different torch can still yield different weights from the same seed.
